### PR TITLE
[GYRO] Refactor gyro driver for dual-gyro support

### PR DESCRIPTION
**Pull request by digitalentity**
Decouple gyro reading and calibration from `gyroUpdate`. This is a pre-requisite for dual-gyro support.

### CODE REVIEW

File: src/main/sensors/gyro.c [hunk 1]
```diff
@@ -81,9 +81,7 @@ FASTRAM gyro_t gyro; // gyro sensor object
 
 STATIC_UNIT_TESTED gyroDev_t gyroDev0;  // Not in FASTRAM since it may hold DMA buffers
 STATIC_FASTRAM int16_t gyroTemperature0;
-
-STATIC_FASTRAM_UNIT_TESTED zeroCalibrationVector_t gyroCalibration;
-STATIC_FASTRAM int32_t gyroADC[XYZ_AXIS_COUNT];
+STATIC_FASTRAM_UNIT_TESTED zeroCalibrationVector_t gyroCalibration0;
```
Comment (DzikuVx) [inline src/main/sensors/gyro.c]: Maybe this could be a 1 element array already?

**Review by DzikuVx — COMMENTED** [review]
@digitalentity LGTM maybe we could store gyro calibration and device as 1 element arrays already?

File: src/main/sensors/gyro.c (hunk 1 above)
Comment (digitalentity) [reply src/main/sensors/gyro.c]: Done

**Comment by digitalentity** [comment]
Before we can have full dual-gyro support the following must be addressed:

1. `detectedSensors` that store detected hardware sensors only have slot for one gyro
2. If gyro fails to detect, FC is bricked and will require a full erase. FC should be allowed to boot without a gyro (or fall back to fake gyro) to allow user to fix the config.
3. Decide how to handle gyro sync. Should we wait for both gyros or should we have a master gyro and read the slave regardless.

**Comment by digitalentity** [comment]
Now FC will boot without a gyro, but will get into hardware failure state.

```
# status
System Uptime: 68 seconds
Current Time: 2041-06-28T01:04:00.000+00:00
Voltage: 1.72V (1S battery - CRITICAL)
CPU Clock=168MHz
STM32 system clocks:
  SYSCLK = 168 MHz
  HCLK   = 168 MHz
  PCLK1  = 42 MHz
  PCLK2  = 84 MHz
Sensor status: GYRO=UNAVAILABLE, ACC=UNAVAILABLE, MAG=NONE, BARO=NONE, RANGEFINDER=NONE, OPFLOW=NONE, GPS=NONE
Stack size: 6144, Stack address: 0x10010000, Heap available: 2048
I2C Errors: 0, config size: 5859, max available config: 131072
ADC channel usage:
   BATTERY : configured = ADC 1, used = ADC 1
      RSSI : configured = ADC 3, used = none
   CURRENT : configured = ADC 2, used = ADC 2
  AIRSPEED : configured = none, used = none
System load: 7, cycle time: 102, PID rate: 9803, RX rate: 49, System rate: 9
Arming disabled flags: ANGLE CAL HWFAIL RX CLI
VTX: not detected
```

**Pull request merged**